opentelemetry/proto/logs/v1/logs.proto

```diff
@@ -0,0 +1,119 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opentelemetry.proto.metrics.v1;
+
+import "opentelemetry/proto/common/v1/common.proto";
+import "opentelemetry/proto/resource/v1/resource.proto";
+
+option java_multiple_files = true;
+option java_package = "io.opentelemetry.proto.logs.v1";
+option java_outer_classname = "LogsProto";
+option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/logs/v1";
+
+// A collection of logs from a Resource.
+message ResourceLogs {
+    // The resource for the spans in this message.
+    // If this field is not set then no resource info is known.
+    opentelemetry.proto.resource.v1.Resource resource = 1;
+
+    // A list of log records.
+    repeated LogRecord logs = 2;
+}
+
+// Possible values for LogRecord.SeverityNumber.
+enum SeverityNumber {
+    UNDEFINED_SEVERITY_NUMBER = 0;
+    TRACE  = 1;
+    TRACE2 = 2;
+    TRACE3 = 3;
+    TRACE4 = 4;
+    DEBUG  = 5;
+    DEBUG2 = 6;
+    DEBUG3 = 7;
+    DEBUG4 = 8;
+    INFO   = 9;
+    INFO2  = 10;
+    INFO3  = 11;
+    INFO4  = 12;
+    WARN   = 13;
+    WARN2  = 14;
+    WARN3  = 15;
+    WARN4  = 16;
+    ERROR  = 17;
+    ERROR2 = 18;
+    ERROR3 = 19;
+    ERROR4 = 20;
+    FATAL  = 21;
+    FATAL2 = 22;
+    FATAL3 = 23;
+    FATAL4 = 24;
+}
+
+// Masks for LogRecord.flags field.
+enum LogRecordFlags {
+    UNDEFINED_LOG_RECORD_FLAG = 0;
+    TRACE_FLAGS_MASK = 0x000000FF;
+}
+
+// A log record according to OpenTelemetry Log Data Model: https://github.com/open-telemetry/oteps/pull/97
+message LogRecord {
+    // time_unix_nano is the time when the event occurred.
+    // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+    // Value of 0 indicates unknown or missing timestamp.
+    fixed64 time_unixnano = 1;
+
+    // A unique identifier for a trace. All logs from the same trace share
+    // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes
+    // is considered invalid. Can be set for logs that are part of request processing
+    // and have an assigned trace id. Optional.
+    bytes trace_id = 2;
+
+    // A unique identifier for a span within a trace, assigned when the span
+    // is created. The ID is an 8-byte array. An ID with all zeroes is considered
+    // invalid. Can be set for logs that are part of a particular processing span.
+    // If span_id is present trace_id SHOULD be also present. Optional.
+    bytes span_id = 3;
+
+    // Optional flags, a bit field. 8 least significant bits are the trace flags as
+    // defined in W3C Trace Context specification. 24 most significant bits are reserved
+    // and must be set to 0. Readers that must not assume that 24 most significant bits
+    // will be zero and must correctly mask the bits when reading 8-bit trace flag (use
+    // flags & TRACE_FLAGS_MASK). Optional.
+    fixed32 flags = 4;
```

Comment:
if we carry flags, should we have a field for `tracestate`?

Reply:
This was discussed in Log Data Model OTEP and at the time it was decided that it is not needed, see https://github.com/open-telemetry/oteps/blob/master/text/0097-log-data-model.md#traceflags-vs-traceparent-and-tracestate
I am open to reconsidering and we can add the tracestate as a field if needed (it will be a backward compatible change, so this PR is still OK to move forward with).

Reply:
minor: the OTEP talks about `TraceFlags` but here we are not using `trace_flags` as I would have naively expected, but just `flags`. is this a previous convention?

Reply:
TraceFlags is a single byte. Protobufs do not have a single byte data type. To avoid wasting an entire fixed32 just for TraceFlags I introduced a general `flags` field where 8 least significant bits are the TraceFlag (as denoted by TRACE_FLAGS_MASK). The 24 most significant bits of `flags` are reserved for future use.

Reply:
got it, thanks for the explanation. I promise to read the comment next time.